internal/filter/on_collab_style.go

```diff
@@ -0,0 +1,17 @@
+package filter
+
+import (
+	"github.com/clean-code-projects/co-coders-api/internal/criteria"
+	"github.com/clean-code-projects/co-coders-api/internal/user"
+)
+
+// OnCollabStyle ..
+func OnCollabStyle(users []user.User, coStyle criteria.CollabStyle) (matchedUsers []user.User) {
+	matchedUsers = []user.User{}
```

Comment:
A note for later.

If we use this style on all the match filters then we will end up looping through the entire user database multiple times. 
We should make note to eventually only loop through the user DB once.

Reply:
I agree with this

Reply:
I also had a thought about this type of filter method. Because it's filtering out all the users that don't match with this specific criteria, if you were to use a chain of filters (in a functional way), the 'filtered out' users would not be present to be matched to other criterias. So, perhaps just returning the matching users might not be enough. 

Instead we might want to consider using a new data structure that contains the user and a list of matches that have occurred. The function would be less of filter and more of a matcher.